### PR TITLE
Convert build.sh into build.js using shelljs

### PR DESCRIPTION
**Pull request by lydell**
For better cross platform compatibility.

### CODE REVIEW

File: scripts/build/build.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  );
+}
+
+shell.echo();
+
+// --- Misc ---
+
+shell.echo("Remove eval");
+shell.sed(
+  "-i",
```
Comment (vjeux) [inline scripts/build/build.js]: This doesn't work on mac. Is it now working?

Comment (vjeux) [inline scripts/build/build.js]: Or windows, or somewhere else. We changed it many times to get it working everywhere. Maybe we should just write it in js now so there aren't compatibility issues?

**Comment by vjeux** [comment]
Thanks! Once the `sed` issue has been figured out, you can merge it :)

File: scripts/build/build.js (hunk 1 above)
Comment (azz) [inline scripts/build/build.js]: Or even just use `os.platform`

Comment (lydell) [inline scripts/build/build.js]: The point of `shelljs` is that it _looks_ like bash, but doesn't use any shell script at all. It's just JavaScript / Node.js APIs behind the scenes. https://github.com/shelljs/shelljs/blob/38b57c8942e68a8bff35fb9177a856dd6f62222a/src/sed.js#L79-L81

**Comment by lydell** [comment]
@vjeux Since the sed thing has been figured out, I guess it's OK to merge?

**Comment by vjeux** [comment]
@lydell yeah, go for it

**Pull request merged**